src/AugLagModel.jl

```diff
@@ -30,36 +31,46 @@ function AugLagModel(model :: AbstractNLPModel, y :: AbstractVector, mu :: Real,
 
   meta = NLPModelMeta(model.meta.nvar, x0 = x0, ncon = ncon, lvar = lvar, uvar = uvar)
 
-  return AugLagModel(meta, Counters(), model, y, mu, x, cx)
+  return AugLagModel(meta, Counters(), model, y, mu, x, cx, y - mu * cx)
 end
 
-function NLPModels.obj(nlp :: AugLagModel, x :: AbstractVector)
-  increment!(nlp, :neval_obj)
+function update_cx!(nlp :: AbstractNLPModel, x :: AbstractVector)
   if x != nlp.x
     cons!(nlp.model, x, nlp.cx)
     nlp.x .= x
+    nlp.muc_y .= nlp.mu .* nlp.cx .- nlp.y
   end
+end
+
+function update_y!(nlp :: AbstractNLPModel)
+  nlp.y .= -nlp.muc_y
+  nlp.muc_y .= nlp.mu .* nlp.cx .- nlp.y
+end
+
+function update_mu!(nlp :: AbstractNLPModel, mu :: Real)
+  nlp.mu = mu
+  nlp.muc_y .= nlp.mu .* nlp.cx .- nlp.y
```

Comment:
Do you mean that we manually update `muc_y` in the algorithm, not in the model?

Reply:
When we call `tron` the value of `x` will change and then `cx` as well, therefore `muc_y` would have to be changed too, right? If I remove the update from inside the function call, `tron` would be called with a fixed `cx`.

Reply:
Something is wrong because in hprod, you call update_cx!, which changes muc_y. Those multipliers must be constant during the inner iterations. Currently, it seems to me that they're changing.

Reply:
`muc_y` are not the multipliers, they are a auxiliary vector storing the `mu * cx - y`. I gather correctly from (4) [here](https://www.researchgate.net/publication/283943976_A_matrix-free_augmented_lagrangian_algorithm_with_application_to_large-scale_structural_design_optimization), then y_k and mu_k are fixed, but `mu_k * c(x) - y_k` will change because `x` changes. At the outer iteration, `muc_y` will be used to update `y` (`update_y!`), but not internally.

Reply:
I see. I got confused between the two. I guess it's ok.

There's a lof of updating of c(x) though.